File: ACM/Acwing/3588/tempCodeRunnerFile.cpp
/*
 * @Author: NEFU AB-IN
 * @Date: 2022-07-30 00:39:09
 * @FilePath: \Acwing\3588\tempCodeRunnerFile.cpp
 * @LastEditTime: 2022-07-30 12:32:14
 */
#include <bits/stdc++.h>
using namespace std;
#define int long long
#define SZ(X) ((int)(X).size())
#define IOS                                                                                                            \
    ios::sync_with_stdio(false);                                                                                       \
    cin.tie(0);                                                                                                        \
    cout.tie(0)
#define DEBUG(X) cout << #X << ": " << X << '\n'
typedef pair<int, int> PII;

const int INF = INT_MAX;
const int N = 1e6 + 10;

signed main()
{
    IOS;
    int n, m;

    function<int(int, int)> cale = [&](int x, int p) {
        int cnt = 0;
        while (x)
            cnt += x /= p;
        return cnt;
    };

    while (cin >> n >> m, n != 0 && m != 0)
    {
        cout << cale(n, 2) - cale(n - m, 2) << '\n';
    }

    return 0;
}

File: ACM/Acwing/tempCodeRunnerFile/tempCodeRunnerFile.py
'''
Author: NEFU AB-IN
Date: 2022-03-07 22:28:25
FilePath: \Acwing\tempCodeRunnerFile.py
LastEditTime: 2022-08-05 21:47:54
'''
N = 1100
MOD = int(1e9 + 7)
dp = [[0] * N for _ in range(N)]  # 表示总和为i, 并且恰好为j个数的方案

n = int(input())
dp[0][0] = 1  #总和为0，恰好0个数的方案有一个

for i in range(1, n + 1):
    for j in range(1, n + 1):
        dp[i][j] = (dp[i - 1][j - 1] + dp[i - j][j]) % MOD

res = 0
for i in range(1, n + 1):
    res = (res + dp[n][i]) % MOD
print(res)